File: C_Programming/Study/Q3_Assignment29.c
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  ChkDigit
// Description:     Assignment_29: Q.3 :Accept a character from user & check whether it is digit or not(0-9).                                       
// Input : 7        
// Output: True
// Input:  d 
// Output: False       
//          
// Date:            15/06/2026
// Author:          Shital Ajit Nikam   
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>
#define TRUE 1
#define FALSE 0

typedef int BOOL;
BOOL ChkDigit(char ch)
{
    BOOL bFlag = FALSE;

    if(ch >= '0' && ch <= '9')
    {
        bFlag = TRUE;
    }
    else
    {
        return bFlag;
    }

    return bFlag;

}

int main()
{
    char cValue = '\0';
    BOOL bRet = FALSE;

    printf("Enter the Digit\n");
    scanf("%c",&cValue);

    bRet = ChkDigit(cValue);

    if(bRet == TRUE)
    {
        printf("It is Digit\n");
    }
    else
    {
        printf("It is not Digit\n");
    }

    return 0;
}

File: Python_Programming/Competative/Q3_Assignment29.py
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
#        
# Description: Assignment_29: Q.3: write a program which accept existing file name through command line arguments.  
#                                  creates new file named Demo.txt & copies all content from given file to Demo.txt
# Input  :  (Command line) ABC.txt       
# Output :  Create Demo.txt & copy contents of ABC.txt inti Demo.txt                                   
#                                     
# Date:            19/07/2026
# Author:          Shital Ajit Nikam   
#
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import os
import sys

#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name: Copy: copy one file content to another.
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

def CopyContent(FileName1):
    try:
        fobj1 = open(FileName1,"r")                  # read mode
        data = fobj1.read()
        
        fobj2 = open("Demo.txt","w")                  # Create file Demo.txt

        fobj2.write(data)
        
        print("File content gets copied to destination file.")         
    
    
    except FileNotFoundError :
        print("Unable to open file, file does not exist.")
        return None            
                

def main():        
        
        Border = "-"*60
        print(Border)                  
                
        if(len(sys.argv) == 2):
            if(sys.argv[1] == "--h" or sys.argv[1] == "--H"):
                print("This script is used to copy the contents of one file into another")
                print("For better usage please check --u flag")
            elif(sys.argv[1] == "--u" or sys.argv[1] == "--U"):
                print("Please execute the script as :")
                print("Python FileName.py FileName")
               
            else:
                              
                CopyContent(sys.argv[1])        
               
                
        else :        
            print("Invalid number of arguments")
            print("Please use --h  / --u for more information")           
                                                 
                   
        print(Border)                    
    
        
if __name__ == "__main__" :
    main()